### ISSUE TITLE
AJP Connector produces NullPointerException

### ISSUE THREAD

**Issue by enisinanaj** [payara/Payara#2649]
# Description #
----------

A NullPointerException occurs when an Apache load balancer is used along with the JK-Connector.

(Same issue is open in Glassfish 5 here: https://github.com/javaee/glassfish/issues/22461)

## Expected Outcome

The expected outcome is that a web application should be able to be accessed through the load balancer which in turn connects to the application server on a given port number.

## Current Outcome

Say I have a load balancer at exposing my applications at the :80 port. I've configured it to have two workers that point to 2 payara 5 application servers at the jk-connector exposed ports. Both listen the port 9010.

When the page http://remoteurl:80/application is accessed from the browser, NullPointerExceptions happen on the AJP side. In particular in the AjpHandlerFilter.

## Steps to reproduce (Only for bug reports) 

1 -**  Make changes to the domain.xml configuration:

        
	<virtual-server network-listeners="http-listener-1,http-listener-2,jk-connector" id="server">
        ...

       
          <protocol name="jk-connector-protocol">
            <http default-virtual-server="server">
              <file-cache></file-cache>
            </http>
          </protocol>

        ...

              
        <network-listeners>
          ...
          <network-listener protocol="jk-connector-protocol" jk-enabled="true" port="8009" name="jk-connector" thread-pool="http-thread-pool" transport="tcp"></network-listener>
        </network-listeners>

## Environment ##

- **Payara Version**: 5
- **Edition**: Full
- **JDK Version**: 8 u162
- **Operating System**: Windows
- **Database**: Oracle

**Comment by ursNiemi** [comment]
Bump. Prevents us from migrating to Payara5.

**Comment by enisinanaj** [comment]
I don't remember what was missing from the previous pull-request. Sorry guys I've had too much work and haven't had the possibility to update the thread.

**Comment by MattGill98** [comment]
Hi @enisinanaj,

No worries at all. Have you submitted a CLA?

Kind regards,

Matt

**Comment by enisinanaj** [comment]
@MattGill98 I have but as you suggested but it doesn't seem to connect

**Comment by MattGill98** [comment]
Hi @enisinanaj,

Do you mean that after making your changes you're unable to connect?

Kind regards,

Matt

**Comment by MattGill98** [comment]
Hi @enisinanaj,

I'm unable to reproduce this issue. Which Payara 5 version are you using? And could you post the contents of your `workers.properties` file?

Kind regards,

Matt

**Comment by smillidge** [comment]
This is not reproducible can you provide a detailed test case?

**Comment by enisinanaj** [comment]
I had the issue with Payara 5.182, Workers file content below.
```
# Define list of workers that will be used
# for mapping requests
worker.list=loadbalancer,status

# Define Node1
# modify the host as your host IP or DNS name.
worker.node1.port=8009
worker.node1.host=193.43.48.152
worker.node1.type=ajp13
worker.node1.connection_pool_size=90
worker.node1.lbfactor=1
worker.node1.socket_keepalive=1
worker.node1.socket_timeout=300
# worker.node1.cachesize=10

worker.node2.port=7209
worker.node2.host=193.43.48.152
worker.node2.type=ajp13
worker.node2.connection_pool_size=90
worker.node2.lbfactor=1
worker.node2.socket_keepalive=1
worker.node2.socket_timeout=300

# Load-balancing behaviour
worker.loadbalancer.type=lb
worker.loadbalancer.balance_workers=node2
worker.loadbalancer.balance_workers=node1
worker.loadbalancer.sticky_session=1
#worker.loadbalancer.method=B
worker.loadbalancer.retries=3

# Status worker for managing load balancer
worker.status.type=status
worker.status.mount=/admin/status

worker.loadbalancer.mount=/webapp
worker.loadbalancer.mount=/webapp/*
worker.loadbalancer.mount=/services/*
```

**Comment by MattGill98** [comment]
Fixed in https://github.com/payara/patched-src-grizzly/pull/9.
